home/import_ordered_content_sets.py

```diff
@@ -0,0 +1,253 @@
+from io import BytesIO
+from logging import getLogger
+from queue import Queue
+
+from django.core.files.base import File  # type: ignore
+from openpyxl import load_workbook
+
+from home.import_helpers import ImportException
+from home.models import ContentPage, OrderedContentSet
+
+logger = getLogger(__name__)
+
+
+@dataclass
+class OrderedContentSetPage:
+    time: str
+    unit: str
+    before_or_after: str
+    page_slug: str
+    contact_field: str
+
+
+class OrderedContentSetImporter:
+    def __init__(
+        self,
+        file: File,
+        file_type: str,
+        progress_queue: Queue[int],
+    ):
+        """
+        Initialize an instance of OrderedContentSetImporter.
+
+        :param file: The file to be imported, as a django.core.files.base.File.
+        :param file_type: The type of the file, e.g. 'CSV' or 'XLSX'.
+        :param progress_queue: A queue.Queue to put progress information on.
+        """
+        self.file = file
+        self.filetype = file_type
+        self.progress_queue = progress_queue
+
+    def _get_or_init_ordered_content_set(
+        self, row: dict[str, str], set_name: str
+    ) -> OrderedContentSet:
+        """
+        Get or initialize an instance of OrderedContentSet from a row of a CSV file.
+
+        :param row: The row of the CSV file, as a dict.
+        :param set_name: The name of the ordered content set.
+        :return: An instance of OrderedContentSet.
+        """
+        ordered_set = OrderedContentSet.objects.filter(name=set_name).first()
+        if not ordered_set:
+            ordered_set = OrderedContentSet(name=set_name)
+
+        return ordered_set
+
+    def _add_profile_fields(
+        self, ordered_set: OrderedContentSet, row: dict[str, str]
+    ) -> None:
+        """
+        Add profile fields to an instance of OrderedContentSet from a row of a CSV file.
+
+        :param ordered_set: An instance of OrderedContentSet.
+        :param row: The row of the CSV file, as a dict.
+        """
+        ordered_set.profile_fields = []
+        for field in [f.strip() for f in (row["Profile Fields"] or "").split(",")]:
+            if field and field != "-":
+                [field_name, field_value] = field.split(":")
+                ordered_set.profile_fields.append((field_name, field_value))
+
+    def _extract_ordered_content_set_pages(
+        self, row: dict[str, str], index: int
+    ) -> list[OrderedContentSetPage]:
+        times = self._csv_to_list(row["Time"])
+        units = self._csv_to_list(row["Unit"])
+        before_or_afters = self._csv_to_list(row["Before Or After"])
+        page_slugs = self._csv_to_list(row["Page Slugs"])
+        # backwards compatiblilty if there's only one contact field
+        contact_fields = row["Contact Field"].split(",")
+        contact_fields = (
+            [p.strip() for p in contact_fields]
+            if len(contact_fields) > 1
+            else [contact_fields[0]] * len(times)
+        )
```

Comment:
I find this a little hard to parse and understand. Also, it looks like we don't `.strip()` the fields if we only have one -- is that intentional?
Perhaps we can make the single vs multiple thing a little more distinct:
```suggestion
        contact_fields = self._csv_to_list(row["Contact Field"])
        # backwards compatiblilty if there's only one contact field
        if len(contact_fields) == 1:
            contact_fields = [contact_fields[0]] * len(times)
```